### PR TITLE
Delivery module new

### PR DESCRIPTION
**Pull request by IroshR**
Delivery service implementation

### CODE REVIEW

File: DeliveryService/src/main/java/com/hemnette/aldi/delivery/config/ProductType.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.hemnette.aldi.delivery.config;
+
+public enum ProductType {
+    SMALL,MEDIUM,LARGE,TWO_MAN_HANDLING,ALL
```
Comment (darshanalafille) [inline DeliveryService/src/main/java/com/hemnette/aldi/delivery/config/ProductType.java:4]: the types were changed little bit, lets be aligned with that.

File: DeliveryService/src/main/java/com/hemnette/aldi/delivery/controller/QuotesController.java [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
+    @Autowired
+    QuotesService quotesService;
 
     @PutMapping("/api/getQuoteForItem")
-    public ResponseEntity<DeliveryQuote> getQuoteForItem(@RequestBody ItemDeliveryQuoteRequest request){
-        return ResponseEntity.ok().build();
+    public ResponseEntity<DeliveryQuoteResponse> getQuoteForItem(@RequestBody ItemDeliveryQuoteRequest request){
+        var result =  quotesService.getQuoteForItem(request);
+
+        return ResponseEntity.ok(result.get());
```
Comment (darshanalafille) [inline DeliveryService/src/main/java/com/hemnette/aldi/delivery/controller/QuotesController.java:22]: It's good coding practice to check whether optional value is present or not.

File: DeliveryService/src/main/java/com/hemnette/aldi/delivery/controller/QuotesController.java [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
     }
 
     @PutMapping("/api/getQuoteForCart")
-    public ResponseEntity<DeliveryQuote> getQuoteForCart(@RequestBody CartDeliveryQuoteRequest request){
-        return ResponseEntity.ok().build();
-    }
+    public ResponseEntity<DeliveryQuoteResponse> getQuoteForCart(@RequestBody CartDeliveryQuoteRequest request){
+        var result =  quotesService.getQuoteForCart(request);
 
+        return ResponseEntity.ok(result.get());
```
Comment (darshanalafille) [inline DeliveryService/src/main/java/com/hemnette/aldi/delivery/controller/QuotesController.java:29]: It's good coding practice to check whether optional value is present or not.

File: DeliveryService/src/main/java/com/hemnette/aldi/delivery/moduleUtil/ItemDeliveryCharge.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package com.hemnette.aldi.delivery.moduleUtil;
+
```
Comment (darshanalafille) [inline DeliveryService/src/main/java/com/hemnette/aldi/delivery/moduleUtil/ItemDeliveryCharge.java:2]: if this class is used as DTO object then better to move to utils module, which is sharable among other modules.

File: DeliveryService/src/main/java/com/hemnette/aldi/delivery/service/QuotesService.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+        ItemDeliveryCharge itemDeliveryCharge = getItemDeliveryCharge(request, location);
+
+        if (itemDeliveryCharge.isDeliverable()){
+            Quote quote = getDeliveryQuote();
+            quote.setTotalPrice(itemDeliveryCharge.getDeliveryCharge());
+
+            return Optional.of(getSuccessDeliveryQuote(quote));
+        }else {
+            return Optional.of(getFailedDeliveryQuote(null));
```
Comment (darshanalafille) [inline DeliveryService/src/main/java/com/hemnette/aldi/delivery/service/QuotesService.java:53]: override the method, don't pass null to methods.

File: DeliveryService/src/main/java/com/hemnette/aldi/delivery/service/QuotesService.java [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                    quote.setTotalPrice(quote.getTotalPrice().add(itemDeliveryCharge.getDeliveryCharge()));
+                }else {
+                    return Optional.of(getFailedDeliveryQuote(null));
+                }
+            }
+
+            return Optional.of(getSuccessDeliveryQuote(quote));
+        }else {
+            logger.info("Empty cart items.");
+            return Optional.of(getFailedDeliveryQuote(null));
```
Comment (darshanalafille) [inline DeliveryService/src/main/java/com/hemnette/aldi/delivery/service/QuotesService.java:89]: overide getFailedDeliveryQuote, dont pass null.

File: MainService/src/main/java/com/hemnette/aldi/controller/DeliveryController.java [hunk 7]
```diff
@@ -41,13 +41,14 @@
             @ApiResponse(code = 404,message = "Not Found")
     })
     public ResponseEntity<DeliveryQuoteResponse> getDeliveryQuoteForItem(HttpServletRequest request, @RequestParam("itemId") Long itemId,
-                                                                         @RequestParam("postCode") String postCode, @RequestParam("suburb") String suburb){
+                                                                         @RequestParam("postCode") String postCode,
+                                                                         @RequestParam("suburb") String suburb){
 
-        if(intrutionDetectionService.isBlackListedIp(CommonUtils.getIp(request)) || !sessionValidator.validateOrigin(request)){
+        /*if(intrutionDetectionService.isBlackListedIp(CommonUtils.getIp(request)) || !sessionValidator.validateOrigin(request)){
```
Comment (darshanalafille) [inline MainService/src/main/java/com/hemnette/aldi/controller/DeliveryController.java:47]: remove commented codes.

**Review by darshanalafille — CHANGES_REQUESTED** [review]

File: DeliveryService/src/main/java/com/hemnette/aldi/delivery/moduleUtil/ItemDeliveryCharge.java (hunk 4 above)
Comment (IroshR) [reply DeliveryService/src/main/java/com/hemnette/aldi/delivery/moduleUtil/ItemDeliveryCharge.java:2]: ItemDeliveryCharge not a DTO object

**Review by darshanalafille — APPROVED** [review]

**Pull request merged**